### PR TITLE
Removed unnecessary `_move_optimizer_state` method overrides

### LINKED ISSUE #10820: Revisit _move_optimizer_state function for all Strategies
## Proposed refactor

From comments in #10596
The device isn't used anymore

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @justusschock @awaelchli @akihironitta

### PR DESCRIPTION
**Pull request by four4fish**
## What does this PR do?

Fixes #10820 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/training_type_plugin.py [hunk 1]
```diff
@@ -108,10 +108,9 @@ def setup_precision_plugin(self) -> None:
 
     def _move_optimizer_state(self, device: Optional[torch.device] = None) -> None:
         """Moves the state of the optimizers to the GPU if needed."""
-        device = device or self.root_device
         for opt in self.optimizers:
             for p, v in opt.state.items():
-                opt.state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, device)
+                opt.state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, device or self.root_device)
```
Comment (four4fish) [inline pytorch_lightning/plugins/training_type/training_type_plugin.py:113]: `device = device or self.root_device` in the beginning of the method will cause issue for TPUSpawning, as the self.root_device called outside the spawning process. So the  `device = device or self.root_device` has to be in `apply_to_collection`

File: pytorch_lightning/plugins/training_type/training_type_plugin.py [hunk 2]
```diff
@@ -108,10 +108,9 @@ def setup_precision_plugin(self) -> None:
 
     def _move_optimizer_state(self, device: Optional[torch.device] = None) -> None:
         """Moves the state of the optimizers to the GPU if needed."""
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/training_type_plugin.py:110]:
```suggestion
        """Moves the state of the optimizers to the appropriate device if needed."""
```

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/training_type_plugin.py (hunk 1 above)
Comment (ananthsub) [reply pytorch_lightning/plugins/training_type/training_type_plugin.py:113]: Could you leave this as a comment in the code? Otherwise it'll be natural to refactor this to be at the start of the method

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/training_type_plugin.py:114]: Yes I'm pretty sure once we wrap Trainer's run method completely with the spawn logic #10059 (step 5), we can call root device here without probelm, and as you say, an opportunity to simplify this method.

**Review by carmocca — APPROVED** [review]
